include/private/node/nettest_wrap.hpp

```diff
@@ -0,0 +1,376 @@
+                    callback->Call(argc, argv);
+                });
+            });
+        });
+    }
+
+    /* clang-format on */
+
+    // ## runners
+
+    // The run method runs the test synchronously. This will block Node until
+    // the test is over. Perhaps not what you want in the common case, but
+    // it may be useful in some specific corner cases.
+    static void run(const Nan::FunctionCallbackInfo<v8::Value> &info) {
+        run_or_start(0, info);
+    }
+
+    // The start method runs the test asynchronously and calls the callback
+    // passed as argument when the test is done. Note that calling this method
+    // will cause Node's event loop to wait for the test to finish, but,
+    // unlike run, start will allow you to do also other things while you're
+    // waiting for the test to finish.
+    static void start(const Nan::FunctionCallbackInfo<v8::Value> &info) {
+        run_or_start(1, info);
+    }
+
+    // ## Internals
+
+  private:
+    // The set_value method is a convenience method. It will make sure
+    // that the number of arguments is the expected one. It also check whether
+    // either run() or start() was already called and throw in such case.
+    static void set_value(int argc,
+            const Nan::FunctionCallbackInfo<v8::Value> &info,
+            std::function<void(NettestWrap *)> &&next) {
+        Nan::HandleScope scope;
+        if (info.Length() != argc) {
+            Nan::ThrowError("invalid number of arguments");
+            return;
+        }
+        next(get_this(info));
+        info.GetReturnValue().Set(info.This());
+    }
+
+    // The get_this() method is a convenience method used by many others to
+    // quickly get the `this` pointer of the class.
+    static NettestWrap *get_this(
+            const Nan::FunctionCallbackInfo<v8::Value> &info) {
+        return ObjectWrap::Unwrap<NettestWrap>(info.Holder());
+    }
+
+    // The run_or_start method implements run() and start().
+    static void run_or_start(
+            int argc, const Nan::FunctionCallbackInfo<v8::Value> &info) {
+        Nan::HandleScope scope;
+        if (info.Length() != argc) {
+            Nan::ThrowError("invalid number of arguments");
+            return;
+        }
+        get_this(info)->nettest.on_destroy([async = get_this(info)->async]() {
+            UvAsyncCtx<>::start_delete(async);
+        });
+        if (argc >= 1) {
+            get_this(info)->nettest.start([
+                async = get_this(info)->async, callback = wrap_callback(info[0])
+            ]() {
+                UvAsyncCtx<>::suspend(async, [callback]() {
+                    Nan::HandleScope scope;
+                    callback->Call(0, nullptr);
+                });
+            });
+        } else {
+            get_this(info)->nettest.run();
+        }
+    }
+
+    // Async is the object used to route MK callbacks to libuv loop.
+    SharedPtr<UvAsyncCtx<>> async;
```

Comment:
I would have maybe named this `ctx` or `uv_asynx_ctx` to avoid clashing with `async` that is in `UvAsyncCtx`.

Reply:
Sure, I can apply this change.